core/kazoo_endpoint/src/kz_endpoint.erl

```diff
@@ -1177,6 +1177,12 @@ build_push_failover(Endpoint, Clid, PushJObj, Call) ->
         ,{<<"Metaflows">>, kz_json:get_value(<<"metaflows">>, Endpoint)}
         ])).
 
+-spec push_headers(kz_json:object()) -> kz_json:object().
+push_headers(PushJObj) ->
+    kz_json:foldl(fun(K, V, Acc) ->
```

Comment:
Sorry I know this is not related but I hate seeing fold called when a map is implemented.
Can you replace this with:
```erlang
push_headers(PushJObj) ->
    kz_json:from_list(
      lists:keymap(fun prefix_header/1, 1, kz_json:to_proplist(PushJObj)).

prepend_header(Header) ->
    <<"X-KAZOO-PUSHER-", Header/binary>>.
```